openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -691,13 +697,17 @@ impl BackingPrivate for TdxBacked {
         let pfns = pfns_handle.base_pfn()..pfns_handle.base_pfn() + pfns_handle.size_pages();
         let overlays: Vec<_> = pfns.collect();
 
-        let flush_page = params
+        let private_pool = params
             .partition
             .private_vis_pages_pool
             .as_ref()
-            .expect("private pool exists for cvm")
+            .expect("private pool exists for cvm");
+
+        let flush_page = private_pool
             .alloc_with_mapping(1.try_into().unwrap(), "tdx_tlb_flush".into())
-            .expect("not out of memory");
+            .unwrap();
```

Comment:
Should we propagate this failure back instead to improve debugability? It's weird to panic since this memory pool has to be sized appropriately and it's very possible we screwed it up.